testing/src/main/java/org/oppia/android/testing/junit/OppiaParameterizedTestRunner.kt

```diff
@@ -0,0 +1,303 @@
+package org.oppia.android.testing.junit
+
+import org.junit.runner.Description
+import org.junit.runner.Runner
+import org.junit.runner.manipulation.Filter
+import org.junit.runner.manipulation.Filterable
+import org.junit.runner.manipulation.Sortable
+import org.junit.runner.manipulation.Sorter
+import org.junit.runner.notification.RunNotifier
+import org.junit.runners.Suite
+import java.lang.annotation.Repeatable
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+
+/**
+ * JUnit test runner that enables support for parameterization, that is, running a single test
+ * multiple times with different data values.
+ *
+ * From a testing correctness perspective, this should only be used to test scenarios of behaviors
+ * that are very similar to one another (i.e. only differ in one or two conditions that can be
+ * data-driven), and that have a large number (i.e. >5) conditions to test. For other situations,
+ * use regular explicit tests, instead (since parameterized tests can hurt test maintainability and
+ * readability).
+ *
+ * This runner behaves like AndroidJUnit4 in that it should work both locally (i.e. via Robolectric)
+ * and on a device (i.e. with Espresso), though the correct Bazel dependency needs to be added based
+ * on the environment in which the test is running.
+ *
+ * To introduce parameterized tests, add this runner along with one or more [Parameter]-annotated
+ * fields and one or more [RunParameterized]-annotated methods (where each method should have
+ * multiple [Iteration]s defined to describe each test iteration). Here's a simple example:
+ *
+ * ```kotlin
+ * @RunWith(OppiaParameterizedTestRunner::class)
+ * class ExampleParameterizedTest {
+ *   @Parameter lateinit var parameter: String
+ *
+ *   @Test
+ *   @RunParameterized(
+ *     Iteration("first", "parameter=first value"),
+ *     Iteration("second", "parameter=second value"),
+ *     Iteration("third", "parameter=third value")
+ *   )
+ *   fun testParams_multipleVals_isConsistent() {
+ *     val result = performOperation(parameter)
+ *     assertThat(result).isEqualTo(consistentExpectedValue)
+ *   }
+ * }
+ * ```
+ *
+ * The test testParams_multipleVals_isConsistent will be run three times, and before each time the
+ * specified parameter value corresponding to each iteration will be injected into the parameter
+ * field for use by the test.
+ *
+ * Note that with Bazel specific iterations can be run by utilizing the test and iteration name,
+ * e.g.:
+ *
+ * ```bash
+ * bazel run //...:ExampleParameterizedTest --test_filter=testParams_multipleVals_isConsistent-first
```

Comment:
Should we add a line explaining `//...:` how does the path will come here?
Currently, it is not mentioned in the [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions#building-the-app) how to run a single test.

Reply:
This feels like something we should fix in the wiki, not in the documentation. The reason I didn't add the fully qualified path is because it could change.

@FareesHussain is this something that we could add to the documentation? Maybe a reference of different useful Bazel command how-tos?

Reply:
Keeping this conversation open for @FareesHussain